compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -713,10 +713,9 @@ impl<'a> Context<'a> {
                                 assert!(!matches!(inline_type, InlineType::Inline), "ICE: Got an ACIR function named {} that should have already been inlined", func.name());
 
                                 let inputs = vecmap(arguments, |arg| self.convert_value(*arg, dfg));
-                                // TODO(https://github.com/noir-lang/noir/issues/4608): handle complex return types from ACIR functions
                                 let output_count =
                                     result_ids.iter().fold(0usize, |sum, result_id| {
-                                        sum + dfg.try_get_array_length(*result_id).unwrap_or(1)
+                                        sum + dfg.type_of_value(*result_id).flattened_size()
```

Comment:
Can you switch this to use `.iter().sum()` instead? I can't suggest it here since that line is unchanged